src/main/java/com/uid2/admin/Main.java

```diff
@@ -195,6 +199,15 @@ public void run() {
             operatorKeyProvider.loadContent(operatorKeyProvider.getMetadata());
             OperatorKeyStoreWriter operatorKeyStoreWriter = new OperatorKeyStoreWriter(operatorKeyProvider, fileManager, jsonWriter, versionGenerator);
 
+            CloudPath s3KeyMetadataPath = new CloudPath(config.getString(Const.Config.S3keysMetadataPathProp));
+            GlobalScope s3KeyGlobalScope = new GlobalScope(s3KeyMetadataPath);
+            RotatingS3KeyProvider s3KeyProvider = new RotatingS3KeyProvider(cloudStorage, s3KeyGlobalScope);
+            S3KeyStoreWriter s3KeyStoreWriter = new S3KeyStoreWriter(s3KeyProvider, fileManager, jsonWriter, versionGenerator, clock, s3KeyGlobalScope);
+            SecureKeyGenerator s3KeyGenerator = new SecureKeyGenerator();
```

Comment:
If this generator is just used by the manager, it should just declare it itself.

Reply:
will do